Assets/BossRoom/Scripts/Server/ServerTestingHotkeys.cs

```diff
@@ -58,6 +65,13 @@ private void Update()
             {
                 NetworkManager.SceneManager.LoadScene("PostGame", LoadSceneMode.Single);
             }
+            if (m_GodModeKeyCode != KeyCode.None && Input.GetKeyDown(m_GodModeKeyCode))
+            {
+                if (m_BossRoomState)
+                {
+                    m_BossRoomState.ToggleGodMode();
+                }
```

Comment:
So this is the first action test/debug hotkey that will now impact several parts of code, and we haven't really discussed how that should be handled compositionally.

In my experience, game-wide toggles like these were kept in a single class wrapped in a preprocessor directive, such that the game could be tested in non-development mode, without any possible test/debug tools influence.

My suggestion here is that we keep `godMode` toggle inside its own `NetworkBehaviour`, since this toggle would only be modified by a server, and game-side classes like `DamageReceiver` would just read from this global value.

@pdeschain thoughts?

Reply:
Good point! I agree this would be a lot cleaner than what I did

Reply:
any reason we're doing god mode game wide and not just per player? That'd simplify this

Reply:
I chose this way since ServerTestingHotkeys is only enabled on host, but I could create a ClientTestingHotkeys that has RPC calls to this, or something similar. I think it would require adding another class in shared with callbacks though

Reply:
That's definitely another approach. Having god-mode per player gives us some flexibility when testing, and it would simplify things. I just figured host would control debug game-wide settings but this works too.

Reply:
I would keep the bool godMode where it is (for now), simply because it's the only cheat we have and there is little reason to architect something more readable and clear then that. The state modification is pretty clean imo.

Reply:
If we're talking architecture - I'd say that considering segregating cheats into a separate class (from game data perspective) is small potatoes in comparison to solving the way our objects resolve their dependencies overall. 

Solving that ^^ would immediately have an effect on how we would segregate our Cheat-manager (and make it a dependency of all the interested classes), our Input manager and our update loop manager, and then wire them all together along with everything else in the game (well, whatever we decide we want to pipe around using this mechanism).

@LPLafontaineB a little nitpick - is it possible for the cheat manager to not be a NetworkBehaviour? I guess not at this moment, as it needs to know isServer to disable itself. I would much prefer classes to not be NetworkBehaviours, or even MonoBehaviours at all if it's possible, but I think that at this point it would require some infrastructure legwork.

Reply:
If I'm understanding correctly, @pdeschain you're saying that I should wait until we decide how and if we want to refactor the global architecture before refactoring the cheat manager, right? If so, I agree.

About your point on NetworkBehaviours and MonoBehaviours, I 100% agree. Here I wanted to keep this PR lightwieght if possible, but if we plan on refactoring the cheats system, we should try to see if it could not be either of those. At this moment, this uses serialized fields for the keycodes of cheats, and the prefabs to spawn, so it might require some thought. I think that would be another task altogether